.github/workflows/ci.yml

```diff
@@ -3,36 +3,279 @@ name: CI
 on:
   workflow_dispatch:
   push:
-#    branches:
-#      - 'main'
-#    paths:
-#      - 'scenario-task'
+    branches:
+      - main
     paths-ignore:
       - '.github/**'
 
 jobs:
-  build:
-    runs-on: ubuntu-latest
-    strategy:
-      fail-fast: false
-    name: CI Build
-    steps:
-      - uses: actions/checkout@v2
-      - uses: actions/setup-java@v2
-        with:
-          distribution: adopt
-          java-version: 8
-          cache: maven
-      - name: Build kafka-samples
-        run: ./mvnw -B package
-        working-directory: kafka-samples
-      - name: Build multio-samples/ticktock
-        run: ./mvnw -B package
-        working-directory: multio-samples/ticktock
-
-#  publish:
-#    needs: [build]
-#    name: Call Publish Snapshot
-#    uses: spring-cloud/spring-cloud-dataflow-samples/.github/workflows/ci-publish.yml@main
-#    secrets:
-#      jfArtifactorySpring: ${{ secrets.JF_ARTIFACTORY_SPRING }}
+
+  kafka-samples:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: ./.github/actions/build-sample-app
+        with:
+          app-dir: 'kafka-samples'
+          mvn-build-commands: '-B clean verify -Pspring'
+          artifactory-publish: true
+          jf-artifactory-spring: ${{ secrets.JF_ARTIFACTORY_SPRING }}
+          docker-push: false
+
+  multi-io-ticktock:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: ./.github/actions/build-sample-app
+        with:
+          app-dir: 'multi-io-samples/ticktock'
+          mvn-build-commands: '-B clean install jib:dockerBuild -pl multi-output-time-rabbit,multi-input-log-rabbit,multi-output-time-kafka,multi-input-log-kafka'
+          artifactory-publish: false
+          docker-push: false
+
+  multi-io-http-purchase-orders:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: ./.github/actions/build-sample-app
+        with:
+          app-dir: 'multi-io-samples/purchase-orders/http-purchase-orders'
+          mvn-build-commands: '-B clean install jib:dockerBuild'
+          artifactory-publish: false
+          docker-push: false
```

Comment:
Disabled until we figure out what to do w/ the Harbor access

Reply:
Spoke w/ @ilayaperumalg and this was initially going to Harbor to deal w/ Dockerhub rate limiting. We will move this back to Dockerhub for now.